presto-common/src/test/java/com/facebook/presto/common/type/TestTimeZoneKey.java

```diff
@@ -47,88 +47,79 @@ public void testUTC()
-        assertSame(TimeZoneKey.getTimeZoneKey("Zulu"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("zulu"), UTC_KEY);
         assertSame(TimeZoneKey.getTimeZoneKey("ZULU"), UTC_KEY);
         assertSame(TimeZoneKey.getTimeZoneKey("UT"), UTC_KEY);
         assertSame(TimeZoneKey.getTimeZoneKey("UCT"), UTC_KEY);
         assertSame(TimeZoneKey.getTimeZoneKey("Universal"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT0"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT+0"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT-0"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT+00:00"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT-00:00"), UTC_KEY);
+        assertTimeZoneNotSupported("GMT+00:00");
+        assertTimeZoneNotSupported("GMT-00:00");
         assertSame(TimeZoneKey.getTimeZoneKey("+00:00"), UTC_KEY);
         assertSame(TimeZoneKey.getTimeZoneKey("-00:00"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("etc/utc"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("etc/gmt"), UTC_KEY);
+        assertTimeZoneNotSupported("etc/utc");
+        assertTimeZoneNotSupported("etc/gmt");
         assertSame(TimeZoneKey.getTimeZoneKey("etc/gmt+0"), UTC_KEY);
         assertSame(TimeZoneKey.getTimeZoneKey("etc/gmt+00:00"), UTC_KEY);
         assertSame(TimeZoneKey.getTimeZoneKey("etc/gmt-00:00"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("etc/ut"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("etc/UT"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("etc/UCT"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("etc/Universal"), UTC_KEY);
+        assertTimeZoneNotSupported("etc/ut");
+        assertTimeZoneNotSupported("etc/UT");
+        assertTimeZoneNotSupported("etc/UCT");
+        assertTimeZoneNotSupported("etc/Universal");
     }
 
     @Test
     public void testHourOffsetZone()
     {
+        assertSame(TimeZoneKey.getTimeZoneKey("GMT"), UTC_KEY);
         assertSame(TimeZoneKey.getTimeZoneKey("GMT0"), UTC_KEY);
         assertSame(TimeZoneKey.getTimeZoneKey("GMT+0"), UTC_KEY);
         assertSame(TimeZoneKey.getTimeZoneKey("GMT-0"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT+0"), UTC_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT-0"), UTC_KEY);
         assertTimeZoneNotSupported("GMT7");
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT+7"), PLUS_7_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT-7"), MINUS_7_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT+7"), PLUS_7_KEY);
-        assertSame(TimeZoneKey.getTimeZoneKey("GMT-7"), MINUS_7_KEY);
+        assertTimeZoneNotSupported("GMT+7");
+        assertTimeZoneNotSupported("GMT-7");
+        assertTimeZoneNotSupported("GMT+7");
+        assertTimeZoneNotSupported("GMT-7");
 
```

Comment:
These should be supported
![Screenshot 2024-01-09 at 10 47 21 AM](https://github.com/prestodb/presto/assets/7980573/737637db-b3a1-487e-8bbb-6ea4a52520a5)

Reply:
I think "GMT+7" is not the right time zone format. The right is "Etc/GMT+7".

Reply:
Yep, you're right!